### DISCUSSION TITLE
Dynamic `url` for binary package

### DISCUSSION THREAD

**Discussion by tillkruss** [Homebrew/discussions#1069] [Tap maintenance and Homebrew development] [answered]
Hello!

We're using Homebrew to install dependencies for a pre-compiled binary. Right now our `url` contains an intel as well as an arm binary for two PHP API versions each.

```
foobar-intel-php74.so
foobar-intel-php80.so
foobar-arm-php74.so
foobar-arm-php80.so
```

Instead of bundling all available binaries for each version, I'd love to download a single binary for the used system.

I noticed casts have an `url do` block, does a similar mechanism exist for formulas?

**Comment by SMillerDev** [comment, marked as answer]
It does not. It would also be incompatible with the shasum.

You can look at the go formula for an example on how to use binaries in formulae.

**Reply by tillkruss** [reply]
@SMillerDev Thanks again, resources is exactly what I was looking for. I wish you had a GH Sponsors button.

**Reply by tillkruss** [reply]
This was extremely helpful as well: https://github.com/SMillerDev/homebrew-tap/blob/main/Formula/php-uuid.rb

**Reply by tillkruss** [reply]
I ended up with something like this:

```ruby
phpv = Formula["php"].version.major_minor

stable do
  url "..."
  sha256 "..."

  resource "ext-relay" do
    if Hardware::CPU.arm?
      if phpv == "8.0"
        url "..."
        sha256 "..."
      elsif phpv == "7.4"
        url "..."
        sha256 "..."
      end
    else 
      if phpv == "8.0"
        url "..."
        sha256 "..."
      elsif phpv == "7.4"
        url "..."
        sha256 "..."
      end
    end
  end
end

head do
    # ...
end
```

**Reply by SMillerDev** [reply]
```rb
phpv = Formula["php"].version.major_minor
```
This will always be 8.0 though since homebrew/core ships with 8.0. You're better off checking the php version manually I think.
